src/sympc/protocol/aby3/aby3.py

```diff
@@ -169,3 +207,78 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
         r_mpc = MPCTensor(shares=r, session=session, shape=x.shape)
         rPrime_mpc = MPCTensor(shares=rPrime, session=session, shape=x.shape)
         return r_mpc, rPrime_mpc
+
+    @staticmethod
+    def local_decomposition(
+        x: ReplicatedSharedTensor, ring_size: str
+    ) -> List[ReplicatedSharedTensor]:
+        """Performs local decomposition to generate shares of shares.
+
+        Args:
+            x(ReplicatedSharedTensor) : input RSTensor.
+            ring_size(str) : Ring size to generate decomposed shares in.
+
+        Returns:
+            List[ReplicatedSharedTensor]: Decomposed shares in the given ring size.
+
+        Raises:
+            ValueError: If RSTensor does not have session uuid.
+        """
+        if x.session_uuid is not None:
+            session = get_session(x.session_uuid)
+        else:
+            raise ValueError("Input RSTensor should have session_uuid")
+
+        ring_size = int(ring_size)
+        tensor_type = get_type_from_ring(ring_size)
+        rank = session.rank
+        nr_parties = session.nr_parties
+
+        zero = torch.zeros(x.shares[0].shape).type(tensor_type)
+
+        shares = [[zero, zero] for i in range(nr_parties)]
```

Comment:
BUG?: 

Careful here. `zero` is the same reference.
Ex:
```
zero = torch.zeros((2,3))
shares = [[zero, zero]]
shares[0][0] = 20
print(shares)
shares
[[tensor([[20.,  0.,  0.], [ 0.,  0.,  0.]]), tensor([[20.,  0.,  0.], [ 0.,  0.,  0.]])]]
```
I am not sure if this is the intended behaviour (I think not).

Reply:
Good catch :+1: .Sure I will do a deep copy